New Commands: 2050 - Call Movement Action

jetrotal opened the pull request:
Here's a bunch of stuff that was already requested in different places.



### Call Movement Action:
```js
@2050("typeOfAction",[targetIsVar,target, parameterIsvar, parameter])
```

A collection of new move commands that can be called as:
```js
@raw 2050, "SetMoveSpeed", targetIsVar,target, speedIsvar, speed
```
```js
@raw 2050, "SetMoveFrequency", targetIsVar,target, frequencyIsvar, frequency
```
```js
@raw 2050, "SetTransparency", targetIsVar,target, transparencyIsvar, transparency
```
```js
@raw 2050, "SetAnimationType", targetIsVar,target, animationTypeIsvar, animationType
```
```js
@raw 2050, "SetFacingLocked", targetIsVar,target, lockedIsvar, locked
```
```js
@raw 2050, "SetLayer", targetIsVar,target, layerIsvar, layer
```
```js
@raw 2050, "SetFlying", targetIsVar,target, charsetIndexIsVar, charsetIndex
```
```js
@raw 2050, "StopMovement", targetIsVar,target
```
```js
@raw 2050, "ChangeCharset/actor1", targetIsVar,target
```
```js
@raw 2050, "Event2Event", eventAIsVar,eventA, eventBisVar, eventB 
//teleports an eventA over eventB, putting EventA to look at same direction as EventB too.
```
```js
@raw 2050, "FaceTowards", eventAIsVar,eventA, eventBisVar, eventB 
```
```jsc
@raw 2050, "FaceAway", eventAIsVar,eventA, eventBisVar, eventB 
```

------------------------------------

### Wait for Single Movement:
~Waits for the end of a Single Event's Movements.~
Already Implement on another PR.

<details>
```js
@raw 2051, "WaitForMovement", 0, 10001, 0, 35, 0, 8
```

```js
@raw 2051, "WaitForMovement", useVarID, ID, useVarTargetVariable, targetVariable, useVarFailuresAmount, failuresAmount
```
While waiting for the Move Route to complete, you can save the outcome of the motion (success or failure) to a targetVariable. Additionally, you can set the number of failed attempts allowed before triggering a failure condition. If the failuresAmount is set to 0, this command will simply wait for the motion to finish without considering failures.
</details>

-------------------------------

old commits comments:
Update Feature - Support Detecting when Move Route Fails Thanks @MackValentine Mack for solving some Core Issues

jetrotal commented:
This one is near completion.
Just need to revamp `@raw 2050, "SetFlying", targetIsVar,target, flyingIsvar, flying`

Once Ghabry have time, he will help refactoring the vehicles code, to allow any event entities to fly or land like airships.

Ghabry commented:
About the semantics of Wait for Single Movement:

I understand this ones:

- Event ID is obviously the event that is checked
- Failures Amount is a threshold to prevent soft locks: 0 = wait infinitely, >=1 abort move route and continue when failure threshold is reached

----

But about the output variable:

The variable is set to 0 when the move route was aborted.

The variable is set to 1 when the move route did not exist or ended successfully.

Is this correct and the intention?

Ghabry committed:
Call Movement Action

`@raw 2050, "SetMoveSpeed", 0, 10001, 1, 3`
@2050("typeOfAction",[targetIsVar,target, parameterIsvar, parameter])

```js
@raw 2051, "", 0, 10001 //Wait for Single Movement
@raw 20140, "Fails to move x times", 8 // start fail branch
@raw 10 //empty space for more cmds
@raw 20141 // end of fail branch
```

Ghabry committed:
FaceTowards, FaceAway, JumpTo And SetAnimationType Commands

```js
@raw 2050, "FaceTowards", eventAIsVar,eventA, eventBisVar, eventB
```
```js
@raw 2050, "FaceAway", eventAIsVar,eventA, eventBisVar, eventB
```

Ghabry committed:
Relocate CallMovementAction to the Map interpreter

florianessl commented:
Small draft proposal for providing extended functionality of "StopMovement" with a new command:
https://github.com/florianessl/Player/commit/07a5b5cfb149de7f969bdf18aa6854cb31d9d7ec
_(Not yet tested - could break)_

Minimum parameter size: **3**
- _param[0]:_ Mode
  - 0: Cancel routes of given events
  - 1: Cancel all active routes except those of the given events
- _param[1]:_ ValueOrVariable modifier (Single value - I see no need to mix modifiers for the subsequent span of parameters)
- _param[2 - x]:_ event ids

**Side note**:
Cancelling move routes for single events is actually possible in Vanilla RPG_RT by just calling "Set Move Route" without providing a list of move_commands. But I'm sure a lot of people aren't aware of this detail.

Ghabry commented:
Yeah I was thinking about this PR here again after seeing "Add Move Route" and imo this command is quite bad. Especially the requirement to have an existing move route (would be also more useful if it would create a new route instead of silently failing).

At least everything that finishes immediately is imo for this command here. Stuff that takes longer to execute needs testing (or is better suited in a move route).

jetrotal commented:
Should we close this?
I already ported my suggestions to #3488, though you guys hate the "add move route" command.

And I saw some people looking for a way to instantly call move route commands, instead of queueing it until next event tick...
